src/Symfony/Component/DomCrawler/AbstractUriElement.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DomCrawler;
+
+/**
+ * Any HTML element that can link to an URI.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+abstract class AbstractUriElement implements UriElementInterface
+{
+    /**
+     * @var \DOMElement
+     */
+    protected $node;
+
+    /**
+     * @var string The method to use for the link
+     */
+    protected $method;
+
+    /**
+     * @var string The URI of the page where the link is embedded (or the base href)
+     */
+    protected $currentUri;
+
+    /**
+     * Constructor.
```

Comment:
@fabpot sure, I see it's present also in `Crawler.php`, `Form.php` and `Field/FormField.php`, should I remove it there too? Or I just edit `AbstractUriElement.php` and `Crawler.php` that are the only ones I changed in this PR?